devops/oci_devops/main.py

```diff
@@ -0,0 +1,57 @@
+from os import environ
+import json
+import oci
+
+
+# Relevant environment variables:
+# REQUIRED:
+#   NAMESPACE: Tenancy namespace
+#   APPLICATION_TARGET: This is the friendly name of your projects "root" compartment.
+#       In this case, it will be "generic-bu"
+#   ALIAS_TARGET: Backwards compatability for some naming stuff I need to refactor to fix
+#       In this case it will be "generic_bu" (the compartment should just be named generic-bu)
+#
+# OPTIONAL:
+#   These two variables help us control authentication so we can support 
+#       multiple use cases - local development and using resource principal auth
+#       when deployed as a container resource
+#   Default is to configure your docker run command so the relevent
+#       .oci/config and api_key resources get mounted from your local file system
+#
+# RESOURCE_PRINCIPAL: Set to true when running container in OCI
+# TENANCY_ID: Required when running in OCI
+
+config = {}
+signer = {}
+tenancy = ""
+
+# This resource principal code is untested yet, placeholder until needed
+if environ.get('RESOURCE_PRINCIPAL') is not None and environ.get('TENANCY_ID') is not None:
```

Comment:
`environ` here is a dict proxy; if the value doesn't exist, running `.get` will raise a KeyError exception; either check for and catch this exception or use the second optional parameter to define the explicit default value 

```suggestion
if environ.get('RESOURCE_PRINCIPAL', None) is not None and environ.get('TENANCY_ID', None) is not None:
```